Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -212,6 +212,15 @@ namespace PhysX
     void EditorHeightfieldColliderComponent::RefreshHeightfield(const AZ::Aabb& dirtyRegion,
         [[maybe_unused]] const Physics::HeightfieldProviderNotifications::HeightfieldChangeMask changeMask)
     {
+        if (changeMask == Physics::HeightfieldProviderNotifications::HeightfieldChangeMask::DestroyBegin ||
+            changeMask == Physics::HeightfieldProviderNotifications::HeightfieldChangeMask::DestroyEnd)
```

Comment:
Nit: Since this is a "mask", it should probably mask and check vs the flag, instead of doing ==.

Reply:
Problem is `HeightfieldChangeMask::Unspecified` is 0xFF which means Destroy flags will be included in there too.
 
It may be better to get rid of Unspecified then and always have something or do a direct listing of flags:
`Unspecified = Settings | HeightData | MaterialData | SurfaceData`

Any suggestions are welcome!

Reply:
I agree, removing it looks way better! 
Updated PR.

Reply:
This is still a mask, so my initial comment still applies.